src/components/Comments.jsx

```diff
@@ -1,17 +1,62 @@
-function Comments({ comments, singleArticle }) {
+import { deleteCommentById, getCommentsByArticleId } from "./utils/utils"; 
+import { useState, useEffect } from "react";
+
+function Comments({ comments, setComments, singleArticle, user }) {
+  const [deletingComment, setDeletingComment] = useState(null);
+
+  useEffect(() => {
+    getComments();
+  }, []); 
+
+  const getComments = () => {
+    getCommentsByArticleId(singleArticle.article_id)
+      .then((data) => {
+        if (data && data.comments) {
+          setComments(data.comments);
+        } 
+      })
+      .catch((error) => {
+        console.error("Error getting comments:", error);
+      });
+  };
+
+  const handleDelete = (comment_id) => {
+    setComments((prevComments) =>
+      prevComments.filter((comment) => comment.comment_id !== comment_id)
+    );
```

Comment:
Since you are optimistically rendering this, please give the user a success or fail message in your then and catch blocks for your API request.